app/main.py

```diff
@@ -1,8 +1,28 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_instances = []
+    for person_data in people:
+        name = person_data["name"]
+        age = person_data["age"]
+        person = Person(name, age)
+        person_instances.append(person)
+    for person_data in people:
+        name = person_data["name"]
+        person_instance = Person.people[name]
+
+        spouse = person_data.get("wife") or person_data.get("husband")
+        if spouse:
+            if "wife" in person_data:
+                person_instance.wife = Person.people[spouse]
+            elif "husband" in person_data:
+                person_instance.husband = Person.people[spouse]
```

Comment:
You are iterating over the `people` list twice. This is inefficient and can be avoided. Consider combining these two loops into one. You can create `Person` instances and set the `wife` and `husband` properties in the same loop.